react-frontend/src/pages/WatchlistPage.js

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect, useContext } from 'react'
+import './styles/WatchlistPage.css'
+import Listing from '../components/Listing/Listing'
+import { fetchUserWatchlist } from '../api'
+import Layout from '../components/Layout/Layout'
+import { AuthContext } from '../../context/AuthContext'
+import { useNavigate } from 'react-router-dom'
+
+const WatchlistPage = () => {
+  const [listings, setListings] = useState([])
+  const [loading, setLoading] = useState(true)
+  const { isAuthenticated } = useContext(AuthContext)
+  const navigate = useNavigate()
+
+  useEffect(() => {
+    const fetchWatchlist = async () => {
+      try {
+        const data = await fetchUserWatchlist()
+        setListings(data)
+      } catch (error) {
+        console.error('Error fetching watchlist:', error)
```

Comment:
Let's add the error state

Reply:
Okay, refactored in  [1745875](https://github.com/YaroslavKSE/AuctionWeb/pull/3/commits/174587596ab873a6ae4b2fbf2caf2a3cd19a0569)